Modules/_asynciomodule.c

```diff
@@ -962,14 +962,18 @@ FutureObj_dealloc(PyObject *self)
 {
     FutureObj *fut = (FutureObj *)self;
 
+    PyObject_GC_UnTrack(self);
+
     if (Future_CheckExact(fut)) {
```

Comment:
nice!  I'll add testcases and add your suggestions, thanks so much for all the help!